// @azure/notification-hubs/dist/commonjs/models/installation.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.
Object.defineProperty(exports, "__esModule", { value: true });
exports.createBrowserInstallation = exports.createWindowsInstallation = exports.createXiaomiInstallation = exports.createFcmV1Installation = exports.createFcmLegacyInstallation = exports.createBaiduInstallation = exports.createAdmInstallation = exports.createAppleInstallation = void 0;
/**
 * Creates an Apple based installation.
 * @param installation - A partial installation used to create the Apple installation.
 * @returns The newly created Apple installation.
 */
function createAppleInstallation(installation) {
    return Object.assign(Object.assign({}, installation), { platform: "apns" });
}
exports.createAppleInstallation = createAppleInstallation;
/**
 * Creates an Amazon Device Messaging (ADM) based installation.
 * @param installation - A partial installation used to create the ADM installation.
 * @returns The newly created ADM installation.
 */
function createAdmInstallation(installation) {
    return Object.assign(Object.assign({}, installation), { platform: "adm" });
}
exports.createAdmInstallation = createAdmInstallation;
/**
 * Creates a Baidu based installation.
 * @param installation - A partial installation used to create the Baidu installation.
 * @returns The newly created Baidu installation.
 */
function createBaiduInstallation(installation) {
    return Object.assign(Object.assign({}, installation), { platform: "baidu" });
}
exports.createBaiduInstallation = createBaiduInstallation;
/**
 * Creates a Firebase legacy HTTP based installation.
 * @param installation - A partial installation used to create the Firebase Legacy HTTP installation.
 * @returns The newly created Baidu installation.
 */
function createFcmLegacyInstallation(installation) {
    return Object.assign(Object.assign({}, installation), { platform: "gcm" });
}
exports.createFcmLegacyInstallation = createFcmLegacyInstallation;
/**
 * Creates an Firebase V1 Cloud Messaging based installation.
 * @param installation - A partial installation used to create the Firebase V1 Cloud Messaging installation.
 * @returns The newly created Firebase V1 Cloud Messaging installation.
 */
function createFcmV1Installation(installation) {
    return Object.assign(Object.assign({}, installation), { platform: "fcmv1" });
}
exports.createFcmV1Installation = createFcmV1Installation;
/**
 * Creates a Xiaomi based installation.
 * @param installation - A partial installation used to create the Xiaomi installation.
 * @returns The newly created Xiaomi installation.
 */
function createXiaomiInstallation(installation) {
    return Object.assign(Object.assign({}, installation), { platform: "xiaomi" });
}
exports.createXiaomiInstallation = createXiaomiInstallation;
/**
 * Creates a Windows Notification Services (WNS) based installation.
 * @param installation - A partial installation used to create the WNS installation.
 * @returns The newly created WNS installation.
 */
function createWindowsInstallation(installation) {
    return Object.assign(Object.assign({}, installation), { platform: "wns" });
}
exports.createWindowsInstallation = createWindowsInstallation;
/**
 * Creates a Web Push based installation.
 * @param installation - A partial installation used to create the Web Push installation.
 * @returns The newly created Web Push installation.
 */
function createBrowserInstallation(installation) {
    return Object.assign(Object.assign({}, installation), { platform: "browser" });
}
exports.createBrowserInstallation = createBrowserInstallation;
