// @azure/notification-hubs/dist/commonjs/models/registration.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.
Object.defineProperty(exports, "__esModule", { value: true });
exports.createXiaomiTemplateRegistrationDescription = exports.createXiaomiRegistrationDescription = exports.createWindowsTemplateRegistrationDescription = exports.createWindowsRegistrationDescription = exports.createFcmV1TemplateRegistrationDescription = exports.createFcmV1RegistrationDescription = exports.createFcmLegacyTemplateRegistrationDescription = exports.createFcmLegacyRegistrationDescription = exports.createBrowserTemplateRegistrationDescription = exports.createBrowserRegistrationDescription = exports.createBaiduTemplateRegistrationDescription = exports.createBaiduRegistrationDescription = exports.createAppleTemplateRegistrationDescription = exports.createAppleRegistrationDescription = exports.createAdmTemplateRegistrationDescription = exports.createAdmRegistrationDescription = void 0;
/**
 * Creates an ADM registration description.
 * @param description - A partial ADM registration description.
 * @returns A created ADM registration description.
 */
function createAdmRegistrationDescription(description) {
    return Object.assign(Object.assign({}, description), { kind: "Adm" });
}
exports.createAdmRegistrationDescription = createAdmRegistrationDescription;
/**
 * Creates an ADM template registration description.
 * @param description - A partial ADM template registration description.
 * @returns A created ADM template registration description.
 */
function createAdmTemplateRegistrationDescription(description) {
    return Object.assign(Object.assign({}, description), { kind: "AdmTemplate" });
}
exports.createAdmTemplateRegistrationDescription = createAdmTemplateRegistrationDescription;
/**
 * Creates an Apple registration description.
 * @param description - A partial Apple registration description.
 * @returns A created Apple registration description.
 */
function createAppleRegistrationDescription(description) {
    return Object.assign(Object.assign({}, description), { kind: "Apple" });
}
exports.createAppleRegistrationDescription = createAppleRegistrationDescription;
/**
 * Creates an Apple template registration description.
 * @param description - A partial Apple template registration description.
 * @returns A created Apple template registration description.
 */
function createAppleTemplateRegistrationDescription(description) {
    return Object.assign(Object.assign({}, description), { kind: "AppleTemplate" });
}
exports.createAppleTemplateRegistrationDescription = createAppleTemplateRegistrationDescription;
/**
 * Creates a Baidu registration description.
 * @param description - A partial Baidu registration description.
 * @returns A created Baidu registration description.
 */
function createBaiduRegistrationDescription(description) {
    return Object.assign(Object.assign({}, description), { kind: "Baidu" });
}
exports.createBaiduRegistrationDescription = createBaiduRegistrationDescription;
/**
 * Creates a Baidu template registration description.
 * @param description - A partial Baidu template registration description.
 * @returns A created Baidu template registration description.
 */
function createBaiduTemplateRegistrationDescription(description) {
    return Object.assign(Object.assign({}, description), { kind: "BaiduTemplate" });
}
exports.createBaiduTemplateRegistrationDescription = createBaiduTemplateRegistrationDescription;
/**
 * Creates a Web Push registration description.
 * @param description - A partial Web Push registration description.
 * @returns A created Web Push registration description.
 */
function createBrowserRegistrationDescription(description) {
    return Object.assign(Object.assign({}, description), { kind: "Browser" });
}
exports.createBrowserRegistrationDescription = createBrowserRegistrationDescription;
/**
 * Creates a Web Push registration description.
 * @param description - A partial Web Push template registration description.
 * @returns A created Web Push template registration description.
 */
function createBrowserTemplateRegistrationDescription(description) {
    return Object.assign(Object.assign({}, description), { kind: "BrowserTemplate" });
}
exports.createBrowserTemplateRegistrationDescription = createBrowserTemplateRegistrationDescription;
/**
 * Creates a Firebase Legacy registration description.
 * @param description - A partial GCM registration description.
 * @returns A created GCM registration description.
 */
function createFcmLegacyRegistrationDescription(description) {
    return Object.assign(Object.assign({}, description), { kind: "Gcm" });
}
exports.createFcmLegacyRegistrationDescription = createFcmLegacyRegistrationDescription;
/**
 * Creates a GCM template registration description.
 * @param description - A partial GCM template registration description.
 * @returns A created GCM template registration description.
 */
function createFcmLegacyTemplateRegistrationDescription(description) {
    return Object.assign(Object.assign({}, description), { kind: "GcmTemplate" });
}
exports.createFcmLegacyTemplateRegistrationDescription = createFcmLegacyTemplateRegistrationDescription;
/**
 * Creates a Firebase V1 registration description.
 * @param description - A partial FCM V1 registration description.
 * @returns A created FCM V1 registration description.
 */
function createFcmV1RegistrationDescription(description) {
    return Object.assign(Object.assign({}, description), { kind: "FcmV1" });
}
exports.createFcmV1RegistrationDescription = createFcmV1RegistrationDescription;
/**
 * Creates a FCM V1 template registration description.
 * @param description - A partial FCM V1 template registration description.
 * @returns A created FCM V1 template registration description.
 */
function createFcmV1TemplateRegistrationDescription(description) {
    return Object.assign(Object.assign({}, description), { kind: "FcmV1Template" });
}
exports.createFcmV1TemplateRegistrationDescription = createFcmV1TemplateRegistrationDescription;
/**
 * Creates a Windows registration description.
 * @param description - A partial Windows registration description.
 * @returns A created Windows registration description.
 */
function createWindowsRegistrationDescription(description) {
    return Object.assign(Object.assign({}, description), { kind: "Windows" });
}
exports.createWindowsRegistrationDescription = createWindowsRegistrationDescription;
/**
 * Creates a Windows template registration description.
 * @param description - A partial Windows template registration description.
 * @returns A created Windows template registration description.
 */
function createWindowsTemplateRegistrationDescription(description) {
    return Object.assign(Object.assign({}, description), { kind: "WindowsTemplate" });
}
exports.createWindowsTemplateRegistrationDescription = createWindowsTemplateRegistrationDescription;
/**
 * Creates a Xiaomi registration description.
 * @param description - A partial Xiaomi registration description.
 * @returns A created Xiaomi registration description.
 */
function createXiaomiRegistrationDescription(description) {
    return Object.assign(Object.assign({}, description), { kind: "Xiaomi" });
}
exports.createXiaomiRegistrationDescription = createXiaomiRegistrationDescription;
/**
 * Creates a Xiaomi template registration description.
 * @param description - A partial Xiaomi template registration description.
 * @returns A created Xiaomi template registration description.
 */
function createXiaomiTemplateRegistrationDescription(description) {
    return Object.assign(Object.assign({}, description), { kind: "XiaomiTemplate" });
}
exports.createXiaomiTemplateRegistrationDescription = createXiaomiTemplateRegistrationDescription;
